pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,168 +46,173 @@ abstract class BuildConfig {
   /// dependency of another. For this it is convenient to know the packageRoot.
   Uri get packageRoot;
 
-  /// The target being compiled for.
-  ///
-  /// Not available in [dryRun].
-  Target get target;
-
   /// The architecture being compiled for.
   ///
-  /// Not available in [dryRun].
-  Architecture get targetArchitecture;
+  /// Not specified (`null`) during a [dryRun].
+  Architecture? get targetArchitecture;
 
   /// The operating system being compiled for.
-  OS get targetOs;
+  OS get targetOS;
 
   /// When compiling for iOS, whether to target device or simulator.
   ///
-  /// Required when [targetOs] equals [OS.iOS].
+  /// Required when [targetOS] equals [OS.iOS].
   ///
-  /// Not available in [dryRun].s
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
```

Comment:
Consider making it throw if `targetOS` is not `OS.iOS` toom and make it non-nullable.
(Unless you expect people to use pattern matching like:
```dart
switch ((c.targetOS, c.targetIOSSdk)) {
   case (OS.iOS, IOSSdk.device) : ...
}
```
then keep it nullable.)